File: 建造者模式/Item.py
class Item(object):
  def Price(self):
    return self.price
  def Name(self):
    return self.name
  def Pack(self):
    self.Pack()

class Meal():
  def __init__(self):
    self.items = []
  def addItem(self, item):
    self.items.append(item)
  def getCost(self):
    cost = 0.0
    for item in self.items:
      cost += item.Price()
    return cost
  def showItems(self):
    print(f"Items:")
    for item in self.items:
      print(f"Name::{item.Name()}, Price::{item.Price()}")

File: 建造者模式/Item.go
package main

import "fmt"

type Item interface {
	Price() float64
	Name() string
	Pack() string
}

type Meal struct {
	Items []Item
}

func (m *Meal) addItem(item Item) {
	m.Items = append(m.Items, item)
}

func (m *Meal) getCost() (sum float64) {
	for _, item := range m.Items {
		sum += item.Price()
	}
	return sum
}

func (m *Meal) showItems() {
	fmt.Println("Items:")
	for _, item := range m.Items {
		fmt.Printf("Item: %s, Price: %.2f, Pack:%s\n", item.Name(), item.Price(), item.Pack())
	}
}
